Extensions/blacklist.js

```diff
@@ -1174,6 +1178,287 @@ XKit.extensions.blacklist = new Object({
+								.filter(value => value instanceof Object)
+							);
+						}
+					}
+
+					return window.tumblr.apiFetch(resource, init);
+				},
+			{ resource, init }
+			);
+		};
+
+		const showNativeExportError = e => {
+			console.error(e);
+			XKit.window.show(
+				'Tumblr Native Filtering Export Error',
+				`<pre>${e.toString()}</pre>`,
+				'error',
+				'<div class="xkit-button default" id="xkit-close-message">OK</div>',
+				true,
+			);
+		};
+
+		const showNativeExport = async () => {
+			const currentFilteredTags = await apiFetch('/v2/user/filtered_tags')
+				.then(({ response: { filteredTags } }) => filteredTags);
+			const currentFilteredContent = await apiFetch('/v2/user/filtered_content')
+				.then(({ response: { filteredContent } }) => filteredContent);
+
+			const blacklistItemData = (XKit.extensions.blacklist.blacklisted || [])
+				.map(name => name.trim())
+				.filter(Boolean)
+				.map(name => {
+					const isTag = name.startsWith('#');
+					const initialValue = name.replace('#', '').replace('*', '');
+
+					const textInput = dom('input', { type: 'text', value: initialValue });
+
+					const tagCheckbox = dom('input', { type: 'checkbox'});
+					const contentCheckbox = dom('input', { type: 'checkbox'});
+					tagCheckbox.checked = isTag;
+					contentCheckbox.checked = !isTag;
+
+					const updateAlreadyFiltered = () => {
+						const isFilteredTag = currentFilteredTags.includes(textInput.value.trim());
+						tagCheckbox.disabled = isFilteredTag;
+						if (isFilteredTag) tagCheckbox.checked = true;
+
+						const isFilteredContent = currentFilteredContent.includes(textInput.value.trim());
+						contentCheckbox.disabled = isFilteredContent;
+						if (isFilteredContent) contentCheckbox.checked = true;
+					};
+					updateAlreadyFiltered();
+					textInput.addEventListener('input', updateAlreadyFiltered);
+
+					return { name, textInput, tagCheckbox, contentCheckbox, updateAlreadyFiltered };
+				});
+
+			if (!blacklistItemData.length) {
+				XKit.window.show(
+					'No Blacklisted Words',
+					"You don't have any blacklisted words to export!",
+					'error',
+					'<div id="xkit-close-message" class="xkit-button">Close</div>',
+				);
+				return;
+			}
+
+			const selectAll = () =>
+				blacklistItemData.forEach(({ tagCheckbox, contentCheckbox, updateAlreadyFiltered }) => {
+					tagCheckbox.checked = true;
+					contentCheckbox.checked = true;
+					updateAlreadyFiltered();
+				});
+
+			const selectNone = () =>
+				blacklistItemData.forEach(({ tagCheckbox, contentCheckbox, updateAlreadyFiltered }) => {
+					tagCheckbox.checked = false;
+					contentCheckbox.checked = false;
+					updateAlreadyFiltered();
+				});
+
+			const doExport = () => {
+				const newTagWords = blacklistItemData
+					.filter(({ tagCheckbox }) => tagCheckbox.checked)
+					.flatMap(({ textInput }) => textInput.value.split(','))
+					.map(word => word.replace('#', '').trim())
+					.filter(word => !currentFilteredTags.includes(word));
```

Comment:
ah, I see, because of the `split` our logic for interpreting this field is out of sync with `isFilteredContent` on line 1285.... feels better to keep those two places in sync to the extent possible, because now we have two different functions interpreting the same string value in two completely different ways (one as an array and one as a string).